homeassistant/components/open_meteo/weather.py

```diff
@@ -43,18 +55,53 @@ def __init__(
         *,
         entry: ConfigEntry,
         coordinator: DataUpdateCoordinator[OpenMeteoForecast],
+        hourly: bool,
     ) -> None:
         """Initialize Open-Meteo weather entity."""
         super().__init__(coordinator=coordinator)
         self._attr_unique_id = entry.entry_id
+        self._config = MappingProxyType(entry.data)
+        self._hourly = hourly
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        name = self._config.get(CONF_NAME)
+        name_appendix = ""
+        if self._hourly:
+            name_appendix = " hourly"
+
+        if name is not None:
+            return f"{name}{name_appendix}"
+
+        return f"{DEFAULT_NAME}{name_appendix}"
 
-        self._attr_device_info = DeviceInfo(
+    @property
+    def unique_id(self) -> str:
+        """Return unique ID."""
+        name_appendix = ""
+        if self._hourly:
+            name_appendix = "-hourly"
+        unique_id = f"{self._attr_unique_id}{name_appendix}"
+        return unique_id
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Device info."""
+        return DeviceInfo(
+            default_name="Forecast",
             entry_type=DeviceEntryType.SERVICE,
-            identifiers={(DOMAIN, entry.entry_id)},
+            identifiers={(DOMAIN, self._attr_unique_id)},  # type: ignore[arg-type]
             manufacturer="Open-Meteo",
-            name=entry.title,
+            model="Forecast",
+            configuration_url="https://open-meteo.com/",
```

Comment:
No reason to change device_info.
Both entities should be same Device (Service)

Reply:
@gjohansson-ST I have moved `DeviceInfo` back to `_attr_device_info` as it was previously.